src/main/java/com/arpnetworking/metrics/mad/Main.java

```diff
@@ -302,6 +316,36 @@ private Injector launchGuice(final ActorSystem actorSystem) {
         return Guice.createInjector(new MainModule(actorSystem, metricsFactory, shutdown));
     }
 
+    @SuppressFBWarnings("REC_CATCH_EXCEPTION")
+    static List<com.arpnetworking.metrics.Sink> createSinks(
+            final ImmutableList<JsonNode> monitoringSinks) {
+        // Until we implement the Commons Builder pattern in the metrics client
+        // library we need to resort to a more brute-force deserialization
+        // style. The benefit of this approach is that it will be forwards
+        // compatible with the Commons Builder approach. The drawbacks are
+        // the ugly way the configuration is passed around (as JsonNode) and
+        // then two-step deserialized.
+        final List<com.arpnetworking.metrics.Sink> sinks = new ArrayList<>();
+        for (final JsonNode sinkNode : monitoringSinks) {
+            @Nullable final JsonNode classNode = sinkNode.get("class");
+            try {
+                if (classNode != null) {
+                    final Class<?> builderClass = Class.forName(classNode.textValue() + "$Builder");
+                    final Object builder = OBJECT_MAPPER.treeToValue(sinkNode, builderClass);
+                    @SuppressWarnings("unchecked")
+                    final com.arpnetworking.metrics.Sink sink =
+                            (com.arpnetworking.metrics.Sink) builderClass.getMethod("build").invoke(builder);
```

Comment:
🤢 

I get why this hack is necessary, it's just really ugly.

Reply:
I started with the preferred approach, but there were some nasty surprises in the metrics client. I put it on hold, and hope to return to it in the future to help undo this hack.